server/src/main/java/org/elasticsearch/common/settings/LocallyMountedSecrets.java

```diff
@@ -191,20 +205,45 @@ LocalFileSecrets processSecretsFile(Path path) throws IOException {
         try (
             var fis = Files.newInputStream(path);
             var bis = new BufferedInputStream(fis);
-            var parser = JSON.xContent().createParser(XContentParserConfiguration.EMPTY, bis)
+            var parser = JSON.xContent()
+                .createParser(XContentParserConfiguration.EMPTY.withDeprecationHandler(DeprecationHandler.IGNORE_DEPRECATIONS), bis)
         ) {
             return secretsParser.apply(parser, null);
         }
     }
 
-    record LocalFileSecrets(Map<String, String> map, ReservedStateVersion metadata) implements Writeable {
+    record LocalFileSecrets(Map<String, byte[]> map, ReservedStateVersion metadata) implements Writeable {
+
+        public static LocalFileSecrets createLocalFileSecrets(
+            Map<String, String> stringSecrets,
+            Map<String, String> fileSecrets,
+            ReservedStateVersion metadata
+        ) {
+            Map<String, byte[]> stringSecretsMap = stringSecrets == null
+                ? Map.of()
+                : stringSecrets.entrySet()
+                    .stream()
+                    .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().getBytes(StandardCharsets.UTF_8)));
+
+            Map<String, byte[]> fileSecretsByteMap = fileSecrets == null
+                ? Map.of()
+                : fileSecrets.entrySet()
+                    .stream()
+                    .collect(Collectors.toMap(Map.Entry::getKey, e -> Base64.getDecoder().decode(e.getValue())));
+
+            Map<String, byte[]> allSecrets = Stream.concat(stringSecretsMap.entrySet().stream(), fileSecretsByteMap.entrySet().stream())
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+            return new LocalFileSecrets(allSecrets, metadata);
+        }
+
         public static LocalFileSecrets readFrom(StreamInput in) throws IOException {
-            return new LocalFileSecrets(in.readMap(StreamInput::readString, StreamInput::readString), ReservedStateVersion.readFrom(in));
+            return new LocalFileSecrets(in.readMap(StreamInput::readString, StreamInput::readByteArray), ReservedStateVersion.readFrom(in));
         }
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
-            out.writeMap((map == null) ? Map.of() : map, StreamOutput::writeString, StreamOutput::writeString);
+            out.writeMap((map == null) ? Map.of() : map, StreamOutput::writeString, StreamOutput::writeByteArray);
```

Comment:
These need to be guarded with version checks right? Or has this not chance of being utilized yet?

Reply:
I didn't think there was any chance of this object being serialized between different node versions, but since I'm not totally sure I should probably add the version checks to be safe.